build/Microsoft.DotNet.Cli.Prepare.targets

```diff
@@ -1,47 +1,98 @@
+  <UsingTask TaskName="PrepareTargets" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="GetCurrentRuntimeInformation" AssemblyFile="$(CLIBuildDll)" />
   <UsingTask TaskName="GenerateBuildVersionInfo" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="CheckPrereqs" AssemblyFile="$(CLIBuildDll)" />
+  <UsingTask TaskName="SetEnvVar" AssemblyFile="$(CLIBuildDll)" />
+
+  <Target DependsOnTargets="BuildDotnetCliBuildFramework" Name="Prepare">
+    <PrepareTargets />
+  </Target>
 
   <Target Name="Init" >
     <!-- Current Runtime Information -->
-    <Rid>
-      <Output TaskParameter="OutputRid" PropertyName="Rid" />
-    </Rid>
-    <Architecture>
-      <Output TaskParameter="OutputArchitecture" PropertyName="Architecture" />
-    </Architecture>
+    <GetCurrentRuntimeInformation>
+      <Output TaskParameter="Rid" PropertyName="Rid" />
+      <Output TaskParameter="Architecture" PropertyName="Architecture" />
+      <Output TaskParameter="OSName" PropertyName="OSName" />
+    </GetCurrentRuntimeInformation>
 
     <!-- Common Properties -->
     <PropertyGroup>
       <BaseOutputDirectory>$(RepoRoot)/artifacts/$(Rid)</BaseOutputDirectory>
       <OutputDirectory>$(BaseOutputDirectory)/stage2</OutputDirectory>
       <Stage2CompilationDirectory>$(BaseOutputDirectory)/stage2compilation</Stage2CompilationDirectory>
       <IntermediateDirectory>$(BaseOutputDirectory)/intermediate</IntermediateDirectory>
+      <PackagesDirectory>$(BaseOutputDirectory)/packages</PackagesDirectory>
     </PropertyGroup>
 
-    <!-- Generates BuildVersion Item
-        BuildVersion Metadata:
-          - Major
-          - Minor
-          - Patch
-          - ReleaseSuffix
-          - CommitHash
-          - CommitCount
-          - VersionSuffix
-          - SimpleVersion
-          - NugetVersion
-          - MsiVersion            -->
     <GenerateBuildVersionInfo RepoRoot="$(RepoRoot)">
-        <Output TaskParameter="OutputBuildVersionInfo"
-                ItemName="BuildVersion" />
+      <Output TaskParameter="VersionMajor" PropertyName="VersionMajor" />
+      <Output TaskParameter="VersionMinor" PropertyName="VersionMinor" />
+      <Output TaskParameter="VersionPatch" PropertyName="VersionPatch" />
+      <Output TaskParameter="CommitHash" PropertyName="CommitHash" />
+      <Output TaskParameter="CommitCount" PropertyName="CommitCount" />
+      <Output TaskParameter="ReleaseSuffix" PropertyName="ReleaseSuffix" />
+      <Output TaskParameter="VersionSuffix" PropertyName="VersionSuffix" />
+      <Output TaskParameter="SimpleVersion" PropertyName="SimpleVersion" />
+      <Output TaskParameter="NugetVersion" PropertyName="NugetVersion" />
+      <Output TaskParameter="MsiVersion" PropertyName="MsiVersion" />
+      <Output TaskParameter="VersionBadgeMoniker" PropertyName="VersionBadgeMoniker" />
+      <Output TaskParameter="Channel" PropertyName="Channel" />
```

Comment:
I also agree on the "move these properties out of the .txt file and into MSBuild", but we can't do that yet.

> For the short term why change all of this from itemgroup metadata to so many properties...

@brthor and I sync'd on this offline.  The main advantage is that the MSBuild code here is self-descriptive of which properties are available (you don't need a comment stating all the custom metadata available on the Item).  Also, since these are all single-valued values, it feels more correct to make these properties, than use an Item.

However, it would be nice if there was some sort of "grouping" mechanism in MSBuild properties (think namespaces), so you could say `BuildVersion_Major`, `BuildVersion_Minor`, etc.  The only grouping mechanism seems to be the underscore, but it doesn't appear to be a standard.  If we find a grouping mechanism in the future, we can change this.